pkg/machine/qemu/machine.go

```diff
@@ -430,45 +409,75 @@ func (v *MachineVM) Init(opts machine.InitOptions) (bool, error) {
 		WritePath: v.getIgnitionFile(),
 		UID:       v.UID,
 	}
-	err = machine.NewIgnitionFile(ign)
+	err := machine.NewIgnitionFile(ign)
 	return err == nil, err
 }
 
-func (v *MachineVM) Set(_ string, opts machine.SetOptions) error {
-	if v.Rootful == opts.Rootful {
-		return nil
-	}
+func (v *MachineVM) Set(_ string, opts machine.SetOptions) ([]error, error) {
+	// If one setting fails to be applied, the others settings will not fail and still be applied.
+	// The setting(s) that failed to be applied will have its errors returned in setErrors
+	var setErrors []error
 
 	state, err := v.State()
 	if err != nil {
-		return err
+		return setErrors, err
 	}
 
 	if state == machine.Running {
 		suffix := ""
 		if v.Name != machine.DefaultMachineName {
 			suffix = " " + v.Name
 		}
-		return errors.Errorf("cannot change setting while the vm is running, run 'podman machine stop%s' first", suffix)
+		return setErrors, errors.Errorf("cannot change settings while the vm is running, run 'podman machine stop%s' first", suffix)
 	}
 
-	changeCon, err := machine.AnyConnectionDefault(v.Name, v.Name+"-root")
-	if err != nil {
-		return err
+	if opts.Rootful != nil && v.Rootful == *opts.Rootful {
+		if err := v.setRootful(*opts.Rootful); err != nil {
+			setErrors = append(setErrors, errors.Wrapf(err, "failed to set rootful option"))
+		} else {
+			v.Rootful = *opts.Rootful
+		}
 	}
 
-	if changeCon {
-		newDefault := v.Name
-		if opts.Rootful {
-			newDefault += "-root"
+	if opts.CPUs != nil && v.CPUs != *opts.CPUs {
+		v.CPUs = *opts.CPUs
+		found := false
+		for i, val := range v.CmdLine {
+			if val == "-smp" {
+				found = true
+				v.CmdLine[i+1] = strconv.Itoa(int(v.CPUs))
+				break
+			}
 		}
-		if err := machine.ChangeDefault(newDefault); err != nil {
-			return err
+		if !found {
+			v.CmdLine = append(v.CmdLine, []string{"-smp", strconv.Itoa(int(v.CPUs))}...)
 		}
 	}
 
-	v.Rootful = opts.Rootful
-	return v.writeConfig()
+	if opts.Memory != nil && v.Memory != *opts.Memory {
+		v.Memory = *opts.Memory
+		found := false
+		for i, val := range v.CmdLine {
+			if val == "-m" {
+				found = true
+				v.CmdLine[i+1] = strconv.Itoa(int(v.Memory))
+				break
+			}
+		}
+		if !found {
+			v.CmdLine = append(v.CmdLine, []string{"-m", strconv.Itoa(int(v.Memory))}...)
+		}
+	}
+
+	if opts.DiskSize != nil && v.DiskSize != *opts.DiskSize {
+		if err := v.resizeDisk(v.DiskSize); err != nil {
```

Comment:
Does this actually resize the filesystem? I suspect that you have to resize the filesystem in the VM. Resizing only the disk is pointless.

Reply:
[core@localhost ~]$ sudo fdisk -l | grep Disk
Disk /dev/vda: 150 GiB, 161061273600 bytes, 314572800 sectors
Disklabel type: gpt
Disk identifier: 23EB28A4-C273-11EC-B7E5-01BDD81662FF
[core@localhost ~]$ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs        4.0M     0  4.0M   0% /dev
tmpfs           979M  100K  979M   1% /dev/shm
tmpfs           392M  5.7M  386M   2% /run
/dev/vda4       150G  2.5G  148G   2% /sysroot
overlay         150G  2.5G  148G   2% /usr
tmpfs           979M     0  979M   0% /tmp
/dev/vda3       362M  110M  230M  33% /boot
tmpfs           196M  8.0K  196M   1% /run/user/501
vol0            229G  113G  117G  50% /Users/acui


Not an expert here but it looks like it resized the fs?